common/common.go

```diff
@@ -182,18 +183,29 @@ func mergeTestSettingsNumberToSetting(testSettingsMap map[string]interface{}, ha
 }
 
 // StartBatchRun starts a batch run or a cross batch run on the server
-func StartBatchRun(urlBase string, apiToken string, organization string, project string, httpHeadersMap map[string]string, testSettingsNumber int, setting string) (*BatchRun, *cli.ExitError) {
+func StartBatchRun(urlBase string, apiToken string, organization string, project string, httpHeadersMap map[string]string, testSettingsNumber int, branchName string, setting string) (*BatchRun, *cli.ExitError) {
 	var testSettings interface{}
 	isCrossBatchRunSetting := (testSettingsNumber != 0)
 	if setting == "" {
-		setting = "{\"test_settings_number\":" + strconv.Itoa(testSettingsNumber) + "}"
+		var branchJsonString = ""
+		if branchName != "" {
+			branchJsonString = ", \"branch_name\":\"" + branchName + "\""
+		}
+		setting = "{\"test_settings_number\":" + strconv.Itoa(testSettingsNumber) + branchJsonString + "}"
 	} else {
 		err := json.Unmarshal([]byte(setting), &testSettings)
 		if err == nil {
 			testSettingsMap, ok := testSettings.(map[string]interface{})
 			if ok {
 				_, hasTestSettings := testSettingsMap["test_settings"]
 				testSettingsNumberInJSON, hasTestSettingsNumber := testSettingsMap["test_settings_number"]
+				testSettingsBranchInJSON, hasTestSettingsBranch := testSettingsMap["branch_name"]
+				if branchName != "" {
+					if hasTestSettingsBranch && branchName != testSettingsBranchInJSON {
+						return nil, cli.NewExitError("--branch_number and --setting have different branch name", 1)
```

Comment:
`branch_number` should be `branch_name`?

Reply:
Thank you!
Fixed in [7a2ba4a](https://github.com/Magic-Pod/magicpod-api-client/pull/28/commits/7a2ba4a70dea613021c54401805397e59b6cab29)